pkg/controllers/node/controller.go

```diff
@@ -150,3 +152,21 @@ func (c *Controller) Register(ctx context.Context, m manager.Manager) error {
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}).
 		Complete(c)
 }
+
+// TriggerTerminationIfExpired is a helper function to be called by other controllers that want to trigger the
+// termination of nodes.
+func TriggerTerminationIfExpired(ctx context.Context, ttlSeconds int64, node *v1.Node, kubeClient client.Client,
```

Comment:
This comment is totally optional and if you disagree with it, I'm happy to merge w/ it as is.

We generally try to avoid static helpers that pass injected dependencies (e.g. kubeclient), parameters only used for logging (e.g. reason), and parameters that are only used by some callers (e.g. return time.Time). Given the relative simplicity of this code if duplicated, being DRY in this case may not be worth it.

Reply:
No strong opinion from me either way! I'll refactor to adhere to the style of the rest of the repo.